apps/emqx_dashboard/test/emqx_dashboard_monitor_SUITE.erl

```diff
@@ -158,14 +158,16 @@ auth_header_() ->
     {"Authorization", "Basic " ++ Basic}.
 
 restart_monitor() ->
-    erlang:exit(erlang:whereis(emqx_dashboard_monitor), killed),
-    ?assertEqual(ok, wait_new_monitor(10)).
-
-wait_new_monitor(Count) when Count =< 0 -> timeout;
-wait_new_monitor(Count) ->
-    case is_pid(erlang:whereis(emqx_dashboard_monitor)) of
+    OldMonitor = erlang:whereis(emqx_dashboard_monitor),
+    erlang:exit(OldMonitor, killed),
+    ?assertEqual(ok, wait_new_monitor(OldMonitor, 10)).
+
+wait_new_monitor(_OldMonitor, Count) when Count =< 0 -> timeout;
+wait_new_monitor(OldMonitor, Count) ->
+    NewMonitor = erlang:whereis(emqx_dashboard_monitor),
+    case is_pid(NewMonitor) andalso NewMonitor =/= OldMonitor of
```

Comment:
Was failing possibly due to a race condition in killing and then checking if the name is registered.

https://github.com/emqx/emqx/runs/5758012844?check_suite_focus=true#step:4:860